### PR TITLE
Document core analysis template report

### LINKED ISSUE #239: Document the core workflow template
**Please provide some background on the proposed additions or changes.**

The template produced by the core workflow is lacking a lot of documentation and explanation of the plots. The template should be updated to include a description of the expected plots. 

**What are the changes that you are proposing?**

In particular a description of the filtering plot and UMAP is needed. We will need to think about how we want to address describing the filtering plot because different methods of filtering can be used.

### PR DESCRIPTION
**Pull request by cbethell**
**Issue Addressed**
Closes #239

**What is the purpose of these changes?**

The core analysis report should be properly documented to give users an idea of how to interpret the plots in the report.

**What changes did you make?**

This PR adds details about the filtering plots, explaining the plots will be different whether `miQC` or `manual` filtering is implemented.

This PR also cleans up the report a bit by removing the numbering before each of the headers and hiding the set up chunks as there are no plots or useful information to be displayed there.

**Any comments, concerns, or questions important for reviewers**
- Does the description above the filtering plots make sense to you? Is it easy to interpret?
- The description above the UMAP plots seems suffice as is (general enough in case users modify the clustering parameters), but please let me know if you feel additional information should be added there.

Here is an example of the report when miQC filtering is implemented:
[library01_core_analysis_report.html.zip](https://github.com/AlexsLemonade/scpca-downstream-analyses/files/10025498/library01_core_analysis_report.html.zip)

Here is an example of the report when manual filtering is implemented:
[library01_core_analysis_report_manual_example.html.zip](https://github.com/AlexsLemonade/scpca-downstream-analyses/files/10025500/library01_core_analysis_report_manual_example.html.zip)

**Checklist**

Place an `x` in all boxes that you have completed.

- [ ] I have run the most recent version of the code
- [ ] If I am adding in reports or generating any plots, I have reviewed the necessary reports
- [x] I have added all necessary documentation (if applicable)

### CODE REVIEW

File: core-analysis/core-analysis-report-template.Rmd [hunk 1]
```diff
@@ -135,6 +135,16 @@ knitr::kable(sample_information, align = 'r') %>%
 
 ### Filtering plots
 
+The filtering plots below represent the cutoffs used to filter and remove low-quality cells.
+When `miQC` filtering is implemented, two plots are displayed.
+The first plot shows the cutoff used for the maximum probability of a cell being compromised.
```
Comment (allyhawkins) [inline core-analysis/core-analysis-report-template.Rmd:140]:
```suggestion
`miQC` relies on fitting a mixture model using the number of genes expressed by a cell and the percentage of mitochondrial reads. 
The first plot will show the model fit by `miQC` with two model fit lines and each cell being colored by the probability of that cell being compromised. 
```
The first plot is actually showing the model produced by miQC and the cells are colored by the probability of being compromised. The second plot is where the cutoff is used to determine if cells are kept or removed.

File: core-analysis/core-analysis-report-template.Rmd [hunk 2]
```diff
@@ -135,6 +135,16 @@ knitr::kable(sample_information, align = 'r') %>%
...
+The second plot indicates which cells have been removed using binary values.
```
Comment (allyhawkins) [inline core-analysis/core-analysis-report-template.Rmd:141]:
```suggestion
The second plot indicates which cells have been removed based on the cutoff used for a cell's probability of being compromised.
Cells labeled with `TRUE` are kept for downstream analysis, while all other cells were removed. 
```
We should make sure to state exactly what the filtering thresholds are and then where they come from.

File: core-analysis/core-analysis-report-template.Rmd [hunk 3]
```diff
@@ -135,6 +135,16 @@ knitr::kable(sample_information, align = 'r') %>%
...
+When `manual` filtering is implemented, one plot is displayed.
+The plot in this case represents the manual filtering cutoffs for maximum percent mitochondrial reads per cell and the minimum number of genes detected per cell.
```
Comment (allyhawkins) [inline core-analysis/core-analysis-report-template.Rmd:143]: I would also say that the lines indicate the cutoffs used and that cells in the upper left quadrant and those with mito content below the max cutoff are kept for further analysis.

**Review by allyhawkins — COMMENTED** [review]
Thanks for doing this, it looks so much better already! I just had some comments on clarifying the filtering plots.

**Comment by cbethell** [comment]
Great points @allyhawkins! I've made the appropriate changes and dug into the plotting functions to ensure that we are indeed giving users the right information to interpret the plots.

Particularly referencing your above comment:
> I would also say that the lines indicate the cutoffs used and that cells in the upper left quadrant and those with mito content below the max cutoff are kept for further analysis.

I double-checked the `plot_manual_filtering()` function in the `filtering-functions.R` script and it looks like I misinterpreted things the first time around, the cutoffs are specific to detected genes and UMI (see below).

Screen Shot 2022-11-18 at 12 31 19 AM

I updated the documentation in the template to reflect this, but please let me know if you have suggestions for alternate wording (or any suggestions otherwise)!

File: core-analysis/core-analysis-report-template.Rmd [hunk 4]
```diff
@@ -135,6 +135,19 @@ knitr::kable(sample_information, align = 'r') %>%
...
 
+The filtering plots below represent the cutoffs used to filter and remove low-quality cells.
+When `miQC` filtering is implemented, two plots are displayed.
+`miQC` relies on fitting a mixture model using the number of genes expressed by a cell and the percentage of mitochondrial reads. 
+The first plot will show the model fit by `miQC` with two model fit lines and each cell being colored by the probability of that cell being compromised.
+The second plot indicates which cells have been removed based on the cutoff used for a cell's probability of being compromised.
+Cells labeled with `TRUE` are kept for downstream analysis, while all other cells were removed.
+When `manual` filtering is implemented, one plot is displayed.
+The plot in this case represents the manual filtering cutoffs for the minimum number of genes detected per cell and the minimum unique molecular identifiers (UMI) per cell.
+The lines on the plot indicate the cutoffs, where cells with number of genes expressed and unique molecular identifiers (UMI) that are both above the minimum cutoffs are kept for downstream analysis.
```
Comment (allyhawkins) [inline core-analysis/core-analysis-report-template.Rmd:146]:
```suggestion
The lines on the plot indicate the cutoffs, where cells with number of genes expressed and unique molecular identifiers (UMI) that are both above the minimum cutoffs are kept for downstream analysis.
Additionally, cells are colored by percent mitochondrial reads. 
Cells with percent mitochondrial reads above the provided cutoff are removed prior to downstream analysis.
```
We also remove cells with high mito content and should state that here.

**Review by allyhawkins — APPROVED** [review]
The additions you made look good regarding the cutoff. I just had one more suggestion on including the mito cutoff, but I don't need to see them again.

**Pull request merged**